src/Cli/dotnet/commands/dotnet-test/SolutionAndProjectUtility.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.Tools.Test;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal static class SolutionAndProjectUtility
+    {
+        public static bool TryGetSolutionOrProjectFilePath(string directory, out string solutionOrProjectFilePath, out bool isSolution)
+        {
+            solutionOrProjectFilePath = string.Empty;
+            isSolution = false;
+
+            // Get all solution files in the specified directory
+            var solutionFiles = GetSolutionFilePaths(directory);
+            // Get all project files in the specified directory
+            var projectFiles = GetProjectFilePaths(directory);
+
+            // If both solution files and project files are found, return false
+            if (solutionFiles.Length > 0 && projectFiles.Length > 0)
+            {
+                VSTestTrace.SafeWriteTrace(() => LocalizableStrings.CmdMultipleProjectOrSolutionFilesErrorMessage);
+                return false;
+            }
+
+            // If exactly one solution file is found, return the solution file path
+            if (solutionFiles.Length == 1)
+            {
+                solutionOrProjectFilePath = solutionFiles[0];
+                isSolution = true;
+                return true;
+            }
+
+            // If exactly one project file is found, return the project file path
+            if (projectFiles.Length == 1)
+            {
+                solutionOrProjectFilePath = projectFiles[0];
+                return true;
+            }
+
+            // If no solution or project files are found, return false
+            VSTestTrace.SafeWriteTrace(() => LocalizableStrings.CmdNoProjectOrSolutionFileErrorMessage);
+            return false;
+        }
+
+        public static IEnumerable<string> GetProjectsFromSolutionFile(string solutionFilePath)
+        {
+            var solutionFile = SolutionFile.Parse(solutionFilePath);
+            return solutionFile.ProjectsInOrder.Select(project => project.AbsolutePath);
+        }
+
+        private static string[] GetSolutionFilePaths(string directory)
+        {
+            var solutionFiles = Directory.GetFiles(directory, "*.sln*", SearchOption.TopDirectoryOnly)
```

Comment:
```suggestion
            var solutionFiles = [
                ..Directory.GetFiles(directory, "*.sln", SearchOption.TopDirectoryOnly),
                ..Directory.GetFiles(directory, "*.slnx", SearchOption.TopDirectoryOnly)
           ]
```

I would suggest doing this instead as to avoid catching `.slnsomething` files

Reply:
With that, you also shouldn't need the `IsSolutionFile` method anymore :)

Reply:
does `dotnet test` need to support solution filters as well?

Reply:
Yes that's currently supported

Reply:
The slnf support will be done in a separate PR.